### ISSUE TITLE
Colorize -printf output when using -color

### ISSUE THREAD

**Issue by bmundt6** [tavianator/bfs#62]
It would be very handy, for instance, if `bfs -type l -color -printf '%f\n'` were to print colorized symlink names.

**Comment by tavianator** [comment]
Agreed!  The main reason I haven't done this yet is that `-printf` supports directives like `%.2f` that `color.c` doesn't handle, and I don't really want to write an entire `printf()` implementation.  But I can sidestep that just by formatting the name with `sprintf()` and then printing it with `cfprintf()`.  Should be ready soon.

**Comment by tavianator** [comment]
The hard case is stuff like `-printf '%10p'` which adds leading padding.  Haven't quite figured out what to do there.
